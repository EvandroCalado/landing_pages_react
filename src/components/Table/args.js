export const args = [
  {
    id: 1,
    head: "title 1",
    body: "content 1",
    footer: "footer 1"
  },
  {
    id: 2,
    head: "title 2",
    body: "content 2",
    footer: "footer 2"
  },
  {
    id: 3,
    head: "title 3",
    body: "content 3",
    footer: "footer 3"
  },
  {
    id: 4,
    head: "title 4",
    body: "content 4",
    footer: "footer 4"
  },
  {
    id: 5,
    head: "title 5",
    body: "content 5",
    footer: "footer 5"
  }
]