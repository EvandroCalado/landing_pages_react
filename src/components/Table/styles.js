import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    overflow-y: auto;
  `}
`;

export const Table = styled.table`
  ${({ theme }) => css`
    width: 100%;
    border-collapse: collapse;
    border: 1px solid #ddd;
    color: ${theme.colors.primaryColor};
    background-color: ${theme.colors.white};
    text-transform: capitalize;

    caption {
      text-align: left;
      font-size: ${theme.font.sizes.small};
    }

    thead,
    tfoot {
      background-color: ${theme.colors.mediumGray};
    }

    td,
    th {
      padding: 0.5rem 1rem;
    }
  `}
`;
