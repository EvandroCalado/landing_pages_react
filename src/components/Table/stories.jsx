import Table from '.';

import { args } from './args';

const TableStories = {
  title: 'Table',
  component: Table,
  args: [args]
};

export const Template = (args) => {
  return (
    <div>
      <Table {...args} />
    </div>
  );
};

export default TableStories