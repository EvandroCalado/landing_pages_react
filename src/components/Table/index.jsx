import P from 'prop-types';
import * as Styled from './styles';

const Table = ({ table = [] }) => {
  return (
    <Styled.Container>
      <Styled.Table>
        <caption>Pricing table</caption>
        <thead>
          <tr>
            {table.map((item) => (
              <th key={item.head}>{item.head}</th>
            ))}
          </tr>
        </thead>
        <tbody>
          <tr>
            {table.map((item) => (
              <td key={item.body}>{item.body}</td>
            ))}
          </tr>
          <tr>
            {table.map((item) => (
              <td key={item.body}>{item.body}</td>
            ))}
          </tr>
          <tr>
            {table.map((item) => (
              <td key={item.body}>{item.body}</td>
            ))}
          </tr>
        </tbody>
        <tfoot>
          <tr>
            {table.map((item) => (
              <td key={item.footer}>{item.footer}</td>
            ))}
          </tr>
        </tfoot>
      </Styled.Table>
    </Styled.Container>
  );
};

Table.propTypes = {
  args: P.objectOf(
    P.shape([
      {
        head: P.string.isRequired,
        body: P.string.isRequired,
        footer: P.string.isRequired,
      },
    ])
  ),
};

export default Table;
