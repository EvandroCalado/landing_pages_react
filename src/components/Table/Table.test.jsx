import Table from '.';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render.theme';

import dataSections from '../../api/data.json';
const section = dataSections[0].sections[5].table;

describe('<Table />', () => {
  it('should render table', () => {
    const {container} = renderTheme(<Table table={section} />);
    expect(screen.getByRole('table')).toBeInTheDocument();
    expect(container).toMatchSnapshot()
  });

  it('should render table without data', () => {
    renderTheme(<Table table={undefined} />);
  });
});
