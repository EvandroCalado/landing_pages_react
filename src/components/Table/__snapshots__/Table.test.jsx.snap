// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render table 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  overflow-y: auto;
}

.c1 {
  width: 100%;
  border-collapse: collapse;
  border: 1px solid #ddd;
  color: #0a1128;
  background-color: #fff;
  text-transform: capitalize;
}

.c1 caption {
  text-align: left;
  font-size: 1.6rem;
}

.c1 thead,
.c1 tfoot {
  background-color: #ddd;
}

.c1 td,
.c1 th {
  padding: 0.5rem 1rem;
}

<div>
  <div
    class="c0"
  >
    <table
      class="c1"
    >
      <caption>
        Pricing table
      </caption>
      <thead>
        <tr>
          <th>
            title 1
          </th>
          <th>
            title 2
          </th>
          <th>
            title 3
          </th>
          <th>
            title 4
          </th>
          <th>
            title 5
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            content1
          </td>
          <td>
            content2
          </td>
          <td>
            content3
          </td>
          <td>
            content4
          </td>
          <td>
            content5
          </td>
        </tr>
        <tr>
          <td>
            content1
          </td>
          <td>
            content2
          </td>
          <td>
            content3
          </td>
          <td>
            content4
          </td>
          <td>
            content5
          </td>
        </tr>
        <tr>
          <td>
            content1
          </td>
          <td>
            content2
          </td>
          <td>
            content3
          </td>
          <td>
            content4
          </td>
          <td>
            content5
          </td>
        </tr>
      </tbody>
      <tfoot>
        <tr>
          <td>
            footer 1
          </td>
          <td>
            footer 2
          </td>
          <td>
            footer 3
          </td>
          <td>
            footer 4
          </td>
          <td>
            footer 5
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;
